### ISSUE TITLE
aria-busy and parent/children required roles

### ISSUE THREAD

**Issue by giacomo-petri** [w3c/aria#2424]
Should we explicitly clarify that required accessible parent/children relationships do not apply to elements with `aria-busy="true"`?

It seems understandable, but I don’t see a place where this is clearly stated. As it stands, tool vendors might still enforce these relationships even when aria-busy is set.

Side note:

I’ve noticed inconsistencies in how browsers handle elements with `aria-busy="true"`. Some treat the role as generic, while others retain the role and announce "busy." What’s the expected behavior? @cookiecrook, would this be a good candidate for WPT?

**Comment by cookiecrook** [comment]
From https://w3c.github.io/aria/#aria-busy
> If it is necessary to make multiple additions, modifications, or removals within a container element that is already either partially or fully rendered, authors **_MAY_** set aria-busy to true on the container element before the first change, and then set it to false when the last change is complete. For example, if multiple changes to a [live region](https://w3c.github.io/aria/#dfn-live-region) should be spoken as a single unit of speech, authors **_MAY_** set aria-busy to true while the changes are being made and then set it to false when the changes are complete and ready to be spoken.

Maybe the change should be:

```diff
- and then set it to false when the last change is complete.
+ and then authors MUST set it to false when the last change is complete.
```

Thoughts?

**Comment by cookiecrook** [comment]
> @cookiecrook, would this be a good candidate for WPT?

I would not think so, because 1) aria-level is not yet testable in WPT, and 2) these are normative author requirements, not user agent requirements.

**Comment by giacomo-petri** [comment]
Sorry, my fault. I was referring to the [last published version](https://www.w3.org/TR/wai-aria/) instead of the draft and focusing on required owned elements. I was questioning whether the aria-busy state removes that requirement, but with the draft version we don't care since we do not longer have required owned elements.

I don’t have a concrete example in mind, but could the opposite be possible? That is, while dynamically updating the DOM, there might be a state where aria-busy is set to true, and some elements require a parent, but no required parent elements are present?

If this scenario is possible (though I’m unsure if it matters), should we clarify that when aria-busy="true", authors are not required to meet this requirement? Also, we should ensure that authors don’t misuse aria-busy simply to pass ARIA validation.

**Comment by cookiecrook** [comment]
> I don’t have a concrete example in mind, but could the opposite be possible? That is, while dynamically updating the DOM, there might be a state where aria-busy is set to true, and some elements require a parent, but no required parent elements are present?

Hypothetically, an orphaned listitem could be outside the DOM tree of a busy list that will pull the listitem into the list using aria-owns, but not into the DOM subtree...

> Also, we should ensure that authors don’t misuse aria-busy simply to pass ARIA validation.

💯
